src/plugins/ui_feature/cell_computed_style.ts

```diff
@@ -0,0 +1,101 @@
+import { LINK_COLOR } from "../../constants";
+import { isObjectEmptyRecursive, removeFalsyAttributes } from "../../helpers/index";
+import {
+  Command,
+  doesCommandInvalidatesTableStyle,
+  invalidateCFEvaluationCommands,
+  invalidateEvaluationCommands,
+} from "../../types";
+import { Border, CellPosition, Style, UID } from "../../types/misc";
+import { UIPlugin } from "../ui_plugin";
+
+export class CellComputedStylePlugin extends UIPlugin {
+  static getters = ["getCellComputedBorder", "getCellComputedStyle"] as const;
+
+  private styles: Record<UID, Record<number, Record<number, Style>>> = {};
+  private borders: Record<UID, Record<number, Record<number, Border | null>>> = {};
+
+  handle(cmd: Command) {
+    if (
+      invalidateEvaluationCommands.has(cmd.type) ||
+      cmd.type === "UPDATE_CELL" ||
+      cmd.type === "EVALUATE_CELLS"
+    ) {
+      this.styles = {};
+      this.borders = {};
+      return;
+    }
+
+    if (doesCommandInvalidatesTableStyle(cmd)) {
+      delete this.styles[cmd.sheetId];
```

Comment:
Is it really needed to remove ALL the styles and border of a sheet when invalidating a table style ? Or is it some kind of security as we don't know the exact table size ? I understand that the add/remove col/row should invalidate everything, but is also the case for the update of a table/filter ?

Reply:
Hiding/unhiding rows will change the table style (banded rows are updated), so UPDATE_FILTER need to invalidate the style.

And after dynamic tables, we do'nt know the actual dimensions of the table added before the evaluation, so we cannot do much better than invalidating the whole sheet